pandas/tests/indexes/test_base.py

```diff
@@ -784,6 +784,69 @@ def test_intersect_str_dates(self):
 
         assert len(result) == 0
 
+    @pytest.mark.parametrize(
+        'fname, sname, expected_name',
+        [
+            ('A', 'A', 'A'),
+            ('A', 'B', None),
+            ('A', None, None),
+            (None, 'B', None),
+            (None, None, None),
+        ])
+    def test_corner_union(self, fname, sname, expected_name):
+        # GH 9943 9862
+        # Test unions with various name combinations
+        # Do not test MultiIndex
+
+        skip_index_keys = ['tuples', 'repeats']
```

Comment:
you can use the indices fixture now (and just skip the ones you don't want)

Reply:
OK, fixed.  Found that `RangeIndex` was not being tested, which meant I had to fix a couple other tests.